.github/workflows/continuous-delivery.yml

```diff
@@ -1,84 +1,96 @@
-        uses: docker/build-push-action@3b5e8027fcad23fda98b2e3ac259d8d67585f671 # v4.0.0
-        with:
-          context: .
-          push: true
-          tags: ${{ steps.meta.outputs.tags }}
-          labels: ${{ steps.meta.outputs.labels }}
-
-      - name: Save image tag for use in deploy job
-        id: save_tag
-        run: echo "::set-output name=tag::${{ steps.package_version.outputs.version }}.${{ github.run_number }}"
-
-  Deploy-Custom:
-    name: Deploy Custom Bot using Docker
-    needs: [Publish]
-    uses: nikolaischunk/utilities/.github/workflows/docker-deploy.yml@main
-    secrets:
-      DOCKER_TAG: 'custom-bot'
-      DOCKER_NETWORK: '${{ secrets.DOCKER_NETWORK }}'
-      DOPPLER_TOKEN: '${{ secrets.DOPPLER_TOKEN_BIRTHDAYY_CUSTOM }}'
-      SSH_HOST: '${{ secrets.SSH_HOST }}'
-      SSH_PRIVATE_KEY: '${{ secrets.SSH_PK }}'
-      SSH_USER: '${{ secrets.SSH_USER }}'
+name: Continuous Delivery
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [testing, production]
+
+permissions:
+  contents: read
+  packages: write
+
+env:
+  # Uses docker.io for Docker Hub if empty
+  REGISTRY: ghcr.io
+  # github.repository as <account>/<repo>
+  IMAGE_NAME: ${{ github.repository }}
+  REPOSITORY_NAME: ${{ github.event.repository.name }}
+
+jobs:
+  Publish:
+    name: Publish image to GHCR 🚀
+    runs-on: ubuntu-latest
+    outputs:
+      deploy_tag: ${{ steps.save_tag.outputs.tag }}
+    steps:
+      - name: Checkout Project
+        uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+          submodules: true
+
+      - name: Extract version from package.json
+        id: package_version
+        uses: saionaro/extract-package-version@v1.2.1
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2.1.0
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata for Docker images
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+          tags: |
+            type=ref,event=branch
+            type=ref,event=pr
+            type=raw,value=${{ steps.package_version.outputs.version }}
+            type=raw,value=${{ steps.package_version.outputs.version }}.${{ github.run_number }}
+            type=raw,value=latest,enable=${{ endsWith(github.ref, github.event.repository.default_branch) }}
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v4.0.0
+        with:
+          context: .
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+
+      - name: Save image tag for use in deploy job
+        id: save_tag
+        run: echo "::set-output name=tag::${{ steps.package_version.outputs.version }}.${{ github.run_number }}"
+  Deploy_Prod:

```

Comment:
What if we want to deploy the current testing branch to production to see if it works?
or when we want to see if the changes made in testing are working for the custom bots? 
I think you did not think this fully throug..

Reply:
It will happen as before, I don't see the problem??

Reply:
launching bots in the different stages to test changes is something rather normal in the business world, especially non production branches like staging or testing. I see the advantage mostly in the Custom Bot deployment.